src/generation.rs

```diff
@@ -293,8 +308,11 @@ pub fn build_octree(
     let octree_meta = &octree::OctreeMeta {
         bounding_box,
         resolution,
+    };
+    let octree_data_provider = OnDiskOctreeDataProvider {
         directory: output_directory.as_ref().to_path_buf(),
     };
+    let octree_data_provider = &octree_data_provider;
```

Comment:
I would rather remove this line and write `&octree_data_provider` wherever this variable is used below.

Reply:
It would only be possible if we cloned the data provider, since below it is moved into another scope, so I will rather go with the reference approach.